### ISSUE TITLE
Update / generated dates/times should be auto added by CoVE

### ISSUE THREAD

**Issue by stevieflow** [OpenDataServices/cove#794]
Context: I'm trying to add the ``generated-datetime`` and ``last-updated-datetime`` to an IATI spreadsheet, for CoVE conversion.  It's "tricky" to do that in a spreadsheet (format wise), but also rather pointless, as it will need updating each and every time.  This doesn't suit the user story of someone pushing data through CoVE a few times, in order to get at a satisfactory file.

I know that on 360 CoVE this data is used in the preview table, but is often missing from the source data

My proposal - we add this data during conversion.  In the case of IATI then ``generated-datetime`` should be a auto generated at the root level for each conversion (do we not record this through the "Data Supplied" function anyway?).  For ``last-updated-datetime``- (which is also the case for 360 ``Last modified``) then this is at the record level.  There might be argument that this should be supplied through source - but it could also make life easier for users to have this a part of the conversion - and then the ability to over ride, if the data is supplied...

NB: I dont know what the situation with OCDS is --> @duncandewhurst ?

**Comment by duncandewhurst** [comment]
We don't auto-fill the equivalent field in the OCDS validator (```publishedDate```).

I think that's mainly because, in OCDS world, we expect most publishers to be producing OCDS from a system rather than hand crafting it, so the validator is a tool to help them get their output right, rather than a tool to produce data for publication.

**Comment by robredpath** [comment]
This would change a fairly basic assumption about CoVE - that is, that if two people convert the same input file, they get the same output file. I'm not saying that's a bad thing, but it's worth noting that it is a departure from something that is currently true. That also changes the assumption that CoVE's conversion function produces a different expression of the same underlying data - fields that were previously missing would be populated. Again, not necessarily a bad thing, but a change. 

If we were to implement this, I wouldn't want to replace the 'pure' conversion functionality, but be an additional feature. If we validate the modified file, we should make it clear that the original file had the data missing, as the ideal is always to fix at source (right?)

I can understand how it makes sense for `generated-datetime` to be automatically populated by CoVE. But `last-updated-datetime` seems to refer more to the individual activities - it feels like that's introducing bad data for the sake of populating a field. `last-update-datetime` is described in the standard as 'The last date/time that the data for this specific activity was updated' - nothing about the data has changed between two consecutive uploads of the file.

**Comment by stevieflow** [comment]
Yeah, I think in this issue I tried to combine multiple standards and different dateTime stamps!  

> I can understand how it makes sense for generated-datetime to be automatically populated by CoVE.

I think it's just that that would be useful to "auto" generate in the IATI CoVE, specifically.  360 have this field (but maybe it should).  The record-level dateTime is a different discussion

Shall I update this issue accordingly?

**Comment by robredpath** [comment]
> Shall I update this issue accordingly?

Yup, if we can make this "CoVE for IATI should automatically add `generated-datetime` for converted files" then I think that's less contentious and clearly defined (and therefore much more likely to happen sooner!)
